File: cjaewon/week01/BOJ_1138_한 줄로 서기.py
"""
단순 구현
"""

import sys

n = int(input())

people = list(map(int, sys.stdin.readline().rstrip().split()))
result = [0 for i in range(n)]

for i, person in enumerate(people):
  # i + 1
  cnt = 0
  
  for j in range(len(result)):
    if cnt == person and result[j] == 0:
      result[j] = i + 1
      break
  
    if result[j] == 0:
      cnt += 1

print(" ".join(map(str, result)))

File: Leeseojun035/week01/BOJ_1138_한 줄로 서기.cpp
#include <iostream>

using namespace std;

int main()
{
    int n;
    cin>>n;
    int *result = new int[n];
    int *input = new int[n];
    for(int i =0; i<n; i++)result[i] = 0;
    for(int i = 0; i<n; i++) cin>>input[i];

    for(int i = 0; i<n; i++){
        int k = 0;// 0 count,남은자리
        int p = 0;//실제 위치
        while(k!=input[i]+1){
            if(result[p]==0){
                k++;
            }
            p++;
        }

        result[p-1]=i+1;
        for(int i=0; i<n; i++){
        cout<<result[i]<<' ';
        }
        cout<<endl;
    }

    for(int i=0; i<n; i++){
        cout<<result[i]<<' ';
        }



    return 0;
}
